src/functional/merge.js

```diff
@@ -69,10 +72,22 @@ export function mergeWithSources(collection, sources, merger) {
 function deepMergerWith(merger) {
   function deepMerger(oldValue, newValue, key) {
     return isDataStructure(oldValue) && isDataStructure(newValue)
-      ? mergeWithSources(oldValue, [newValue], deepMerger)
+      ? areMergeable(oldValue, newValue)
+        ? mergeWithSources(oldValue, [newValue], deepMerger)
+        : newValue
```

Comment:
Curious what you think about calling the custom merger in this scenario? Previous logic is to do this

```suggestion
      && areMergeable(oldValue, newValue)
      ? mergeWithSources(oldValue, [newValue], deepMerger)
```

Reply:
Yeah, I definitely considered it. My main thought was that it would be an additional breaking change since the user would need to handle more cases (although I can't imagine it would have that big of an impact), but it would provide more flexibility for the user. I'm in favor of it if we're not concerned about it being a breaking change.

Reply:
Yeah the domain in which we're changing behavior is already in the weird cases. My thought was that it would be easier to explain the behavior of the function. Essentially both before and after the change we could correctly say "if we can merge em we do, if not we call your merge conflict function which has a default of `(old, new) => new`

Reply:
I think we also have some license to make very modest breaking changes in the name of fixing bad behavior since the plan is land pre final 4.0

Reply:
Updated.